### PR TITLE
refactor(ivy): remove LNode.tNode

### PR DESCRIPTION
**Pull request by kara**
This PR removes `LNode.tNode`. 

Notes:
- It migrates many of our main utility functions to using `TNode` args instead of `LNode` args where possible, adding as an additional argument where `LNodes` would be difficult to look up.  
- There is now a `getTNode(index)` helper that allows lookup via index in `TView.data`.
- `LViewData[HOST_NODE]` now stores `TNode` instances instead of `LNode` instances. There is a new `getHostElementNode()` helper in `node_manipulation.ts` to get component host `LNodes` where necessary.
- There's also some minor refactoring to share more code where I saw some duplication while re-wiring args. e.g. `createRootComponent`, `getHighestElementContainer`, `getContainerRenderParent`

### CODE REVIEW

File: packages/core/test/bundling/hello_world/bundle.golden_symbols.json [hunk 1]
```diff
@@ -224,12 +239,18 @@
   {
     "name": "getParentLNode"
   },
+  {
+    "name": "getParentOrContainerNode"
```
Comment (kara) [inline packages/core/test/bundling/hello_world/bundle.golden_symbols.json]: This is brought in by `createLNode` and brings in 3 or 4 other small functions. I think we will be able to remove as soon as we do the injector refactor.

**Review by mhevery — APPROVED** [review]

File: packages/core/test/bundling/hello_world/bundle.golden_symbols.json (hunk 1 above)
Comment (mhevery) [reply packages/core/test/bundling/hello_world/bundle.golden_symbols.json]: SGTM

File: packages/core/src/render3/component_ref.ts [hunk 2]
```diff
@@ -115,70 +115,60 @@ export class ComponentFactory<T> extends viewEngine_ComponentFactory<T> {
...
-                createLNode(++index, TNodeType.Element, nodeList[j] as RElement, null, null);
-            if (previousTNode) {
-              previousTNode.next = lNode.tNode;
-            } else {
-              firstTNode = lNode.tNode;
-            }
-            previousTNode = lNode.tNode;
+            const tNode =
+                createNodeAtIndex(++index, TNodeType.Element, nodeList[j] as RElement, null, null);
+            previousTNode ? (previousTNode.next = tNode) : (firstTNode = tNode);
```
Comment (JoostK) [inline packages/core/src/render3/component_ref.ts]: Why was this rewritten like this? The former code seems easier to read.

File: packages/core/src/render3/di.ts [hunk 3]
```diff
@@ -754,13 +743,13 @@ class ViewContainerRef extends viewEngine_ViewContainerRef {
 
   remove(index?: number): void {
     const adjustedIdx = this._adjustIndex(index, -1);
-    removeView(this._lContainerNode, adjustedIdx);
+    removeView(this._lContainerNode, this._tContainerNode as TContainerNode, adjustedIdx);
```
Comment (JoostK) [inline packages/core/src/render3/di.ts]: Looks like the cast isn't necessary?

**Comment by kara** [comment]
[presubmit](http://test/OCL:213006395:BASE:213006465:1536946878573:e4cb7bf2)

**Pull request closed without merging**